### PR TITLE
replaced print statements with log

### PR DESCRIPTION
**Pull request by shimwell**
Currently GEOUNED prints quite a lot to the terminal

However we could move all of that terminal output to a logger

The logger then writes the print message along with date, time, filename, line number and other options to a log file

This includes more information than a print statement, it is persistent and searchable

The ```verbose``` options is now redundant so this has also been removed

Here is an example log file that gets produced automatically

```bash
(geouned_py311) j@mini:~/GEOUNED$ cat geouned.log
2024-05-08 09:58:03,181 :: INFO :: start :: 358 :: start
2024-05-08 09:58:03,181 :: INFO :: start :: 360 :: GEOUNED version 1.0.1 19/03/2024
FreeCAD version 0.21.2
2024-05-08 09:58:03,181 :: INFO :: start :: 393 :: read step file : /home/j/GEOUNED/testing/inputSTEP/biseau.stp
2024-05-08 09:58:03,198 :: INFO :: start :: 398 :: End of loading phase
2024-05-08 09:58:03,199 :: INFO :: start :: 400 :: 0:00:00.017430
2024-05-08 09:58:03,199 :: INFO :: decompose_solids :: 602 :: Decomposing solid: 1/1
2024-05-08 09:58:03,344 :: INFO :: start :: 447 :: End of decomposition phase
2024-05-08 09:58:03,344 :: INFO :: start :: 454 :: Building cell: 1
2024-05-08 09:58:03,347 :: INFO :: start :: 476 :: 0:00:00.148343
2024-05-08 09:58:03,347 :: INFO :: start :: 494 :: Build Void
2024-05-08 09:58:03,347 :: INFO :: start :: 495 :: []
2024-05-08 09:58:03,347 :: INFO :: void_generation :: 44 :: Build Void highest enclosure
2024-05-08 09:58:03,348 :: INFO :: get_void_def :: 96 :: Loop, Box to Split :{iloop}, {nvoid}
2024-05-08 09:58:03,348 :: INFO :: get_void_def :: 121 :: build complementary 1 0
2024-05-08 09:58:03,349 :: INFO :: start :: 530 :: build Time: 2024-05-08 09:58:03.349553 - 2024-05-08 09:58:03.347509
2024-05-08 09:58:03,349 :: INFO :: start :: 532 :: 0:00:00.167967
2024-05-08 09:58:03,349 :: INFO :: write_input :: 75 :: write MCNP file /home/j/GEOUNED/tests_outputs/testing/inputSTEP/biseau/biseau.mcnp
2024-05-08 09:58:03,350 :: INFO :: write_xml :: 28 :: write OpenMC xml file /home/j/GEOUNED/tests_outputs/testing/inputSTEP/biseau/biseau.xml
2024-05-08 09:58:03,350 :: INFO :: write_py :: 94 :: write OpenMC python script /home/j/GEOUNED/tests_outputs/testing/inputSTEP/biseau/biseau.py
2024-05-08 09:58:03,350 :: INFO :: write_input :: 66 :: write Serpent file /home/j/GEOUNED/tests_outputs/testing/inputSTEP/biseau/biseau.serp
2024-05-08 09:58:03,350 :: INFO :: write_phits :: 61 :: write PHITS file /home/j/GEOUNED/tests_outputs/testing/inputSTEP/biseau/biseau.inp
2024-05-08 09:58:03,351 :: INFO :: __write_phits_cell_block__ :: 167 :: Unified the inner void cell(s) definition
2024-05-08 09:58:03,351 :: INFO :: start :: 587 :: End of MCNP, OpenMC, Serpent and PHITS translation phase
2024-05-08 09:58:03,351 :: INFO :: start :: 589 :: Process finished
2024-05-08 09:58:03,351 :: INFO :: start :: 590 :: 0:00:00.169686
2024-05-08 09:58:03,351 :: INFO :: start :: 592 :: Translation time of solid cells 2024-05-08 09:58:03.347509 - 2024-05-08 09:58:03.199163
2024-05-08 09:58:03,351 :: INFO :: start :: 593 :: Translation time of void cells 2024-05-08 09:58:03.349553 - 2024-05-08 09:58:03.347509
```

The output can be customised to include more details with each print statement if we like.

### CODE REVIEW

**Comment by alberto743** [comment]
Sure. You can assign the review to me. I can look at it tomorrow or on Friday.

**Comment by psauvan** [comment]
Its is fine to move all the terminal output into a log file, but I think that we should keep some of them so the user can check the current status of the process, Is there any library implementing progress bar in python?

**Comment by psauvan** [comment]
yes, this would be nice

File: src/geouned/GEOUNED/__init__.py
Comment (psauvan) [inline src/geouned/GEOUNED/__init__.py:1]: at line 451 change the loop by 
```python
for i, m in enumerate(tqdm(MetaList, desc="translating solid cells")):
```

**Review by psauvan — CHANGES_REQUESTED** [review]
see comment https://github.com/GEOUNED-org/GEOUNED/pull/133/commits/60c5c9e86b786935c0ef607faf67a63d23b1ec4e#r1594069544

**Comment by alberto743** [comment]
Hi @shimwell . Thank you very much for the development.
The branch is fine for merging for me, but I do have the following comments
1. the `logging` should be used also for other log files, e.g. `fuzzySurfaces`. This may be described inside another another issue.
2. While I like the `tqdm` module for the progress bar, I personally do not agree with @psauvan on the fact that the core module should dump anything on the standard output/error. Instead, any information written to stdout/stderr may arrive from some calling scripts but should not be part of the core progressing, which should interact with other modules exclusively via the stack of function calls or via appropriate `logging` calls. Again, this may be tracked via another issue.

**Comment by alberto743** [comment]
The new issues to track the suggestions have been opened.

**Pull request merged**